WalletWasabi.Gui/Controls/WalletExplorer/SendTabViewModel.cs

```diff
@@ -80,9 +82,25 @@ public string PayjoinEndPoint
 
 		private IPayjoinClient GetPayjoinClient()
 		{
-			if (!string.IsNullOrWhiteSpace(PayjoinEndPoint) && Uri.IsWellFormedUriString(PayjoinEndPoint, UriKind.Absolute))
+			if (!string.IsNullOrWhiteSpace(PayjoinEndPoint) &&
+				Uri.IsWellFormedUriString(PayjoinEndPoint, UriKind.Absolute))
 			{
 				var payjoinEndPointUri = new Uri(PayjoinEndPoint);
+				if (!Global.Config.UseTor)
+				{
+					if (payjoinEndPointUri.DnsSafeHost.EndsWith(".onion", StringComparison.OrdinalIgnoreCase))
+					{
+						Logger.LogWarning("Payjoin server is a hidden service but Tor is disabled. Ignoring...");
+						return null;
+					}
+
+					if (Global.Config.Network != Network.RegTest && payjoinEndPointUri.Scheme != "https")
+					{
+						Logger.LogWarning("Payjoin server is not using https. Ignoring...");
```

Comment:
It should say that it is not using neither onion or https. I don't know how this is expressed in English but I think should be in the log.

```suggestion
						Logger.LogWarning("Payjoin server is not exposed as onion hidden service nor https. Ignoring...");
```

Reply:
Meh, not sure about this, but if it makes you happy 😄